clients/client-py/taskcluster/helper.py

```diff
@@ -0,0 +1,128 @@
+
+import os
+
+import logging
+import json
+from taskcluster.generated import _client_importer
+
+logger = logging.getLogger(__name__)
+
+
+class TaskclusterConfig(object):
+    """
+    Local configuration used to access Taskcluster service and objects
+    """
+
+    def __init__(self, url="https://community-tc.services.mozilla.com"):
+        self.options = None
+        self.secrets = None
+        self.default_url = os.environ.get("TASKCLUSTER_ROOT_URL", url)
+
+    def auth(self, client_id=None, access_token=None, max_retries=12):
+        """
+        Build Taskcluster credentials options
+        Supports, by order of preference:
+         * directly provided credentials
+         * credentials from local configuration
+         * credentials from environment variables
+         * taskclusterProxy
+         * no authentication
+        """
+        self.options = {"maxRetries": max_retries}
+
+        if client_id is None and access_token is None:
+            # Credentials preference: Use local config from release-services
+            xdg = os.path.expanduser(os.environ.get("XDG_CONFIG_HOME", "~/.config"))
+            config = os.path.join(xdg, "taskcluster", "config.json")
+            logger.debug("Reading local configuration in {}".format(config))
+            try:
+                assert os.path.exists(config), "No user config available"
+                data = json.load(open(config))
+                client_id = data["auth"]["client_id"]
+                access_token = data["auth"]["access_token"]
+                assert (
+                    client_id is not None and access_token is not None
+                ), "Missing values in user folder"
+                logger.info("Using taskcluster credentials from local configuration")
+            except Exception:
+                # Credentials preference: Use env. variables
+                client_id = os.environ.get("TASKCLUSTER_CLIENT_ID")
+                access_token = os.environ.get("TASKCLUSTER_ACCESS_TOKEN")
+                logger.info("Using taskcluster credentials from environment")
+        else:
+            logger.info("Using taskcluster credentials from cli")
+
+        if client_id is not None and access_token is not None:
+            # Use provided credentials
+            self.options["credentials"] = {
+                "clientId": client_id,
+                "accessToken": access_token,
+            }
+            self.options["rootUrl"] = self.default_url
+
+        elif "TASK_ID" in os.environ:
+            # Use Taskcluster Proxy when running in a task
+            logger.info("Taskcluster Proxy enabled")
+            self.options["rootUrl"] = "http://taskcluster"
+
+        else:
+            logger.info("No Taskcluster authentication.")
+            self.options["rootUrl"] = self.default_url
+
+    def get_service(self, service_name):
+        """
+        Build a Taskcluster service instance using current authentication
```

Comment:
This specifically returns synchronous services -- could you add support for async as well?